### PR TITLE
mds/MDSRank: Add set_history_slow_op_size_and_threshold for op_tracker

### PR DESCRIPTION
**Pull request by YiteGu**
Fixes: https://tracker.ceph.com/issues/61749

```
void OpHistory::_insert_delayed(const utime_t& now, TrackedOpRef op) {
...
  double opduration = op->get_duration();
...
  if (opduration >= history_slow_op_threshold.load()) {
    slow_op.insert(make_pair(op->get_initiated(), op));
    logger->inc(l_trackedop_slow_op_count);
  }
...
}
```
mds op_tracker have no set slow op threshold, and history_slow_op_threshold default value is 0, cause to mds have slow op alway.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by YiteGu** [comment]
@rzarzynski have a code review

**Comment by YiteGu** [comment]
@batrick help me code review, if you have time

**Review by joscollin — CHANGES_REQUESTED** [review]
https://tracker.ceph.com/projects/cephfs 
Please create a tracker ticket, mentioning the issue and attach to this PR.

**Comment by YiteGu** [comment]
Fixes: https://tracker.ceph.com/issues/61749

File: src/common/options/mds.yaml.in [hunk 1]
```diff
@@ -1166,6 +1166,24 @@ options:
...
+  with_legacy: true
+# Track the op if over this threshold
+- name: mds_op_history_slow_op_threshold
+  type: uint
+  level: advanced
+  desc: track the op if over this threshold
+  default: 10
+  services:
+  - mds
+  with_legacy: true
```
Comment (dparmar18) [inline src/common/options/mds.yaml.in:1186]: there are `osd_op_history_slow_op_size` and `osd_op_history_slow_op_threshold` in `global.yaml.in` while `mon.yaml.in` also have these values, these values are set in `OSD::OSD()` and `Monitor::Monitor()` respectively, these should be sufficient to set them. Although you mentioned in commit message "mds op_tracker have no set slow op threshold, and history_slow_op_threshold default value is 0, cause to mds have slow op alway." Is it some sort of bottleneck with MDS you're facing due to this?

**Review by batrick — APPROVED** [review]

File: src/common/options/mds.yaml.in (hunk 1 above)
Comment (YiteGu) [reply src/common/options/mds.yaml.in:1186]: We have slow op counter in TrackedOp, you can see: https://github.com/ceph/ceph/pull/52068. If we no set this value, this metrics will continuously growing.

Comment (dparmar18) [reply src/common/options/mds.yaml.in:1186]: Hmm, Interesting. So, `op_history_slow_op_threshold` and `op_history_slow_op_size` are part of OSD and Monitor, and with this PR, MDS too. Let's say all of them set different values for `op_history_slow_op_threshold` and `op_history_slow_op_size`, in this case do we have any overriding mechanism? Else it is possible that a daemon's values might affect another, right?

Comment (YiteGu) [reply src/common/options/mds.yaml.in:1186]: mds, osd, mon and every ceph daemon have self op_tracker object, each op_tracker manage their own configurations without affecting each other

Comment (dparmar18) [reply src/common/options/mds.yaml.in:1186]: Ah, got it.

Comment (dparmar18) [reply src/common/options/mds.yaml.in:1186]: I had a misconception, nvm

**Review by vshankar — APPROVED** [review]
Good catch!

**Comment by vshankar** [comment]
@rishabh-d-dave please pick this in your next integration run.

**Review by rishabh-d-dave — APPROVED** [review]
Testing was successful - https://tracker.ceph.com/projects/cephfs/wiki/Main#13-Jul-2023

**Pull request merged**